lib/promscrape/discovery/docker/container.go

```diff
@@ -67,14 +69,51 @@ func parseContainers(data []byte) ([]container, error) {
 	return containers, nil
 }
 
-func addContainersLabels(containers []container, networkLabels map[string]*promutils.Labels, defaultPort int, hostNetworkingHost string) []*promutils.Labels {
+func addContainersLabels(containers []container, networkLabels map[string]*promutils.Labels, defaultPort int, hostNetworkingHost string, matchFirstNetwork bool) []*promutils.Labels {
+	allContainers := make(map[string]container)
+	for _, c := range containers {
+		allContainers[c.ID] = c
+	}
+
 	var ms []*promutils.Labels
 	for i := range containers {
 		c := &containers[i]
 		if len(c.Names) == 0 {
 			continue
 		}
-		for _, n := range c.NetworkSettings.Networks {
+
+		networks := c.NetworkSettings.Networks
+		networkMode := c.HostConfig.NetworkMode
+		if len(networks) == 0 {
+			// Try to lookup shared networks
+			for {
```

Comment:
I tried to understand purpose of this `for` loop and it seems not needed to me. 

We can just check the reference at `connected` container:

```
    containerID, ok := containerID(networkMode)
    if ok{
      tmpContainer, exist := allContainers[containerID]
      if exist{
        networks = tmpContainer.NetworkSettings.Networks    
      }
    }
```

Reply:
It's possible that the connected container will connect to another container.
```json
[{
	"container_id": "1",
	"network": "some network"
}, {
	"container_id": "2",
	"network": "container:1"
}, {
	"container_id": "3",
	"network": "container:2"
}]
```